api/app/src/mappings/garmin/body-composition.ts

```diff
@@ -0,0 +1,81 @@
+import { Body } from "@metriport/api";
+import { groupBy } from "lodash";
+import { z } from "zod";
+import {
+  DataType,
+  garminMetaSchema,
+  garminTypes as t,
+  User,
+  UserData,
+} from ".";
+import { PROVIDER_GARMIN } from "../../shared/constants";
+import { toISODate } from "../../shared/date";
+
+export const mapToBody = (
+  items: GarminBodyCompositionList
+): UserData<Body>[] => {
+  const type: DataType = "body";
+  const byUAT = groupBy(items, (a) => a.userAccessToken);
+  return Object.entries(byUAT).flatMap(([key, values]) => {
+    const uat = key;
+    const userData = values;
+    const user: User = {
+      userAccessToken: uat,
+    };
+    return userData.map(garminBodyCompositionToBody).map((data) => ({
+      user,
+      typedData: { type, data, x: data.bone_mass_kg },
+    }));
+  });
+};
+
+export const garminBodyCompositionToBody = (
+  gBody: GarminBodyComposition
+): Body => {
+  const res: Body = {
+    metadata: {
+      // TODO https://github.com/metriport/metriport-internal/issues/166
+      date: toISODate(gBody.measurementTimeInSeconds),
+      source: PROVIDER_GARMIN,
+    },
+  };
+  if (gBody.muscleMassInGrams != null) {
+    res.muscle_mass_kg = gBody.muscleMassInGrams / 1_000;
+  }
+  if (gBody.boneMassInGrams != null) {
+    res.bone_mass_kg = gBody.boneMassInGrams / 1_000;
+  }
+  if (gBody.bodyFatInPercent != null) {
+    res.body_fat_pct = gBody.bodyFatInPercent;
+  }
+  if (gBody.weightInGrams != null) {
+    res.weight_kg = gBody.weightInGrams / 1_000;
```

Comment:
we're making use of a convert library for this elsewhere in the code, can use so we don't have to hardcode conversions every time:

https://github.com/metriport/metriport/blob/0bd80ebeb7b80eaa1f544f1c4bb49f77bbab9923/api/app/src/mappings/fitbit/body.ts#L18

Reply:
That's nice!